addons/udes_stock/models/stock_location.py

```diff
@@ -312,3 +312,11 @@ def get_available_stock_locations(self):
          their children) stock available for fulfilling orders. Should be
         overridden where necessary"""
         return self.env.ref("stock.stock_location_stock")
+
+    def get_supplied_location_is_child(self, location):
```

Comment:
Would suggest to transfer this in base models, we may need to check for user templates, products etc. Maybe the name should be more generic example `record_is_child(self, child) `

Reply:
Good idea, not sure on the name though (I didn't like `get_supplied_location_is_child` either).

The problem with `record_is_child` is that it implies that we are just checking if the record in `self` is a child of something else. But instead we are checking that `self` is a parent of the supplied record.

Maybe `record_is_parent`?

Reply:
I think you are better with names, definitely :)

Reply:
I wish that were true, I've added some very curious names in the code!

Reply:
I think... `record_is_child_of`... that way it is clear from the name that we are checking it is a child of a particular location. :)

Reply:
Went for `record_is_child_of_self` and added unit tests, please review.